mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala

```diff
@@ -52,8 +51,20 @@ private[spark] object CholeskyDecomposition {
   def inverse(UAi: Array[Double], k: Int): Array[Double] = {
     val info = new intW(0)
     lapack.dpptri("U", k, UAi, info)
-    val code = info.`val`
-    assert(code == 0, s"lapack.dpptri returned $code.")
+    checkReturnValue(info, "dpptri")
     UAi
   }
+
+  private def checkReturnValue(info: intW, method: String): Unit = {
+    info.`val` match {
+      case code if code < 0 =>
+        throw new IllegalStateException(s"LAPACK.$method returned $code; arg ${-code} is illegal")
```

Comment:
Should we output ```arg ${-code}```? Since the arguments is internal used and MLlib users have no sense of them and can do nothing to verify it. If the arguments of ```dppsv``` is illegal, it should be caused by bugs of MLlib.